provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -474,16 +479,31 @@ func createClassicClusterObject(ctx context.Context,
 	builder.CloudProvider(cmv1.NewCloudProvider().ID(awsCloudProvider))
 	builder.Product(cmv1.NewProduct().ID(rosaProduct))
 	builder.Region(cmv1.NewCloudRegion().ID(state.CloudRegion.ValueString()))
-	multiAZ := common.Bool(state.MultiAZ)
+	multiAZ := common.BoolWithFalseDefault(state.MultiAZ)
 	builder.MultiAZ(multiAZ)
+
 	// Set default properties
 	properties := make(map[string]string)
 	for k, v := range OCMProperties {
 		properties[k] = v
 	}
 	if !state.Properties.IsUnknown() && !state.Properties.IsNull() {
-		for k, v := range state.Properties.Elements() {
-			properties[k] = v.(types.String).ValueString()
+		propertiesElements := make(map[string]types.String, len(state.Properties.Elements()))
+		d := state.Properties.ElementsAs(ctx, &propertiesElements, false)
```

Comment:
maybe we can have it as function? I assume it is sort of validation that map is map[string]string, right?